docs/blueprints/EP025.rst

```diff
@@ -0,0 +1,111 @@
+
+
+Motivation
+==========
+
+A loop on a switch interface can be intentionally or not. Typically, intentional loops are for re-injecting packets in the switch pipeline to perform another packet transformation. On the other hand, unintentional loops can lead to major dataplane problems, especially if that interface has significant network traffic being being forwarded. A loop detection mechanism via LLDP will empower operators to define certain actions to be taken when a loop happens.
+
+
+Specification
+=============
+
+It is desirable to support ignored loops configuration that can be changed dynamically via API. Since this feature is relatively small and wouldn't require much extra entries or states, then using `topology` switches' and interfaces' metadata storage would simplify the implementation and maintenance since currently adding support to a new storage with storehouse requires a significant effort development-wise. In the future, this can be migrated to another storage (collection) when an alternative is available. The following metadata would be used:
+
+- ``switches.<dpid>.metadata.ignored_loops``:
+
+This is for configuring which loops should be ignored on a ``<dpid>``, the value is a list of list pair of interface port numbers. For instance, if it's desirable to ignore a loop between interface number 5 and 6, and also between port 1 and itself: ``[[5, 6], [1,1]]``
+
+- ``interfaces.<interface_id>.metadata.looped``:
+
+This ``looped`` attribute is for state management, it will contain the following object whenever a loop is in place:
+
+.. code-block:: python3
+
+   {
+     'port_numbers': [<int>, <int>]
+     'detected_at': <str>, in this format %Y-%m-%dT%H:%M:%S
+   }
+
+Looped states should be removed if the interface has been disabled or deactivated, also there should be a periodic routine to remove the ``looped`` metadata interface state if the loop hasn't been detected anymore, this clean up routine should run at a lower pace than of_lldp polling, usually 3-4 times slower, and if the loop hasn't been seen within this time frame again then it would assume that the loop has stopped.
+
+The following options would be configurable on settings:
+
+- ``LLDP_LOOP_ACTIONS``:
+
+This is for configuring which actions should be taken once a loop is found. The actions are a list of strings, ``log`` and ``disable`` are both supported. The ``log`` action will log an warning message informing that a loop was detected on a dpid and on which interfaces. The ``disable`` action will disable an interface via the ``topology`` NApp, and eventually enable it again if the loop stops. For instance, ``['log', 'disable']`` would apply both actions to any loop that isn't ignored and is found. The default actions would be ``['log']`` based on what's expected to be used in production in this first iteration.
```

Comment:
@jab1982 indeed, it's confusing when I read this again, the intention here of `disable` would be to disable via the `topology` NApp using this endpoint `/api/kytos/topology/v3/interfaces/<intf_id>/disable`, and the `enable` again part would be to undo the disable, so calling the `/api/kytos/topology/v3/interfaces/<intf_id>/enable` to enable it again once the loop has stopped. These topology endpoints, at the moment, they don't send a `PortMod` [and an issue has been mapped recently for this](https://github.com/kytos-ng/topology/issues/72), so the disable/enable would affect the `enable` state, which currently it would imply that pathfinder wouldn't be able to find a path there, and mef_eline ideally should react too, but there's a missing topology event in this endpoint as well  and it's only sending `kytos/topology.updated` (that mef_eline doesn't subscribe to), so EVCs wouldn't converge, however, once we have that issue 72 prioritized it should also enable the full blown usage. Does that clarify the understanding? I'll augment the phrase with more information.

Reply:
@jab1982 updated this paragraph on this commit https://github.com/kytos-ng/kytos/pull/181/commits/141b0f8b776b455f8f781d36de2c4ac89f7fdec3, let me know if it's less confusing now.